fio/lib/src/utils.ts

```
@@ -0,0 +1,281 @@
+  publicKey,
+  actionData,
+  account,
+  action,
+  fioBaseUrl,
+}: CreateRawTxParams): Promise<{
+  rawTx: any;
+  serializedActionData: string;
+  chainId: string;
+}> => {
+  const chainData: ChainData = await getChainData(fioBaseUrl);
+  Transactions.FioProvider = {
+    accountHash: Fio.accountHash,
+  };
+
+  const { fioTokenAbi } = await getAbiProvider(account);
+  Transactions.abiMap.set(fioTokenAbi.account_name, fioTokenAbi);
+
+  const serializedActionData = await createSerializeActionData({
+    account,
+    actionName: action,
+    actionData,
+    chainId: chainData.chain_id,
+  });
+
+  const transaction = new Transactions();
+  const rawTx = await transaction.createRawTransaction({
+    action: action,
+    account: account,
+    data: actionData,
+    chainData: chainData,
+    publicKey: publicKey,
+  });
+
+  return {
+    rawTx,
+    serializedActionData,
+    chainId: chainData.chain_id,
+  };
+};
+
+/**
+ * Serializes action data.
+ * @param account The account.
+ * @param actionName The action name.
+ * @param actionData The action data.
+ * @param chainId The chain ID.
+ * @returns The serialized action data.
+ */
+const createSerializeActionData = async ({
+  account,
+  actionName,
+  actionData,
+  chainId,
+}: CreateSerializeActionDataParams): Promise<string> => {
+  const { abiProvider } = await getAbiProvider(account);
+  // @ts-ignore
+  const fioApi = new Api({
+    chainId: chainId,
+    abiProvider: abiProvider,
+  });
+  const contract = await fioApi.getContract(account);
+  // @ts-ignore
+  const serializedActionData = serializeActionData(
+    contract,
+    account,
+    actionName,
+    actionData
+  );
+  return serializedActionData;
+};
+
+/**
+ * Gets the ABI provider and FIO token ABI.
+ * @param account The account.
+ * @returns The FIO token ABI and ABI provider.
+ */
+const getAbiProvider = async (
+  account: string
+): Promise<{ fioTokenAbi: FioTokenAbi; abiProvider: AbiProvider }> => {
```

we should throw an error if we don't have ABI for the given account in the args.

---

Done